src/preload/preload.c

```diff
@@ -685,7 +703,7 @@ static void __attribute__((constructor)) init_process(void) {
 
   buffer_enabled = !!getenv(SYSCALLBUF_ENABLED_ENV_VAR);
 
-  pthread_atfork(NULL, NULL, post_fork_child);
+  signal(SYSCALLBUF_DESCHED_SIGNAL, sig_init_thread);
```

Comment:
Ah, right, forgot about that.  I'll add a comment and we can worry about it if it ever comes up.